Library/Homebrew/system_command.rb

```diff
@@ -87,6 +88,7 @@ def initialize(
     executable,
     args: [],
     sudo: false,
+    sudo_user: nil,
```

Comment:
Don't think that `with_sudo_user` makes sense here: that's what the `sudo` parameter does already. 

Making `root` a default for `sudo_user` might skim the diff a little bit. And it's a better default.

Reply:
Changing the default of `sudo_user` to `root` did the opposite of skimming the diff: there are far fewer places in Homebrew where "root" is required.

The only other approach I can think of is to allow the `sudo` parameter to accept both boolean and string arguments:

- `true`: use sudo without `--user`
- `false`: don't use sudo
- "<somebody>": use sudo with `--user <somebody>`
- "": use sudo without `--user`
- nil: don't use sudo

Reply:
@Kentzo I'll be more direct: the API as-is here is confusing and not good enough. I am open to landing this PR but not with the API as-is.

> But it's an extra argument already, may as well make it more flexible and in line with options of the sudo command.

No, let's not make it more flexible. Flexible APIs where you don't exercise that flexibility are bad. [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).

`sudo_user` is never set to a non-nil, non-`"root"` value. As a result: it does not make any sense to me for it to be `T.nilable(String)` when it could be a boolean and simplify `sudo_user.nil?` checks into boolean checks.

> Changing the default of sudo_user to root did the opposite of skimming the diff: there are far fewer places in Homebrew where "root" is required.

Ok, that's fine, it can be left as-is. Please just change this argument to a boolean with a different name as desired.

Thanks!

Reply:
`sudo_as_root` has more elegant. `runas` is weird as it runs two words together.